### ISSUE TITLE
InstanceSegmentation generator

### ISSUE THREAD

**Issue by 0x00b1** [broadinstitute/keras-rcnn#3]
keras-rcnn should include an InstanceSegmentation generator like Keras’s [ImageDataGenerator](https://keras.io/preprocessing/image/#imagedatagenerator).

**Comment by 0x00b1** [comment]
The ImageSegmentationGenerator should be an easier task.

The dictionary should look similar to the existing ObjectDetectionGenerator dictionary except each box includes a `mask` key that has the mask pathname, e.g. 

```Python
{
    "boxes": [
        {
            "x_minimum": 0,
            "y_minimum": 0,
            "x_maximum": 0,
            "y_maximum": 0,
            "description: "foo",
            "mask: "mask.png"
        }
    ],
    "c": 224,
    "channels": 3
    "pathname": "image.png",
    "r": 224
}
```

It should yield masks (in addition to an image, bounding boxes, and labels), e.g.

```Python
generator = ImageSegementationGenerator()

generator = generator.flow(training)

image, (boxes, labels, masks) = generator.next()
```
